sw/airborne/modules/nav/nav_survey_rectangle_rotorcraft.c

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright (C) 2007-2009  ENAC, Pascal Brisset, Antoine Drouin
+ *                    2015  NAC-VA, Eduardo Lavratti
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+/**
+ * @file modules/nav/nav_survey_rectangle_rotorcraft.c
+ *
+ * Automatic survey of a rectangle for rotorcraft.
+ *
+ * Rectangle is defined by two points, sweep can be south-north or west-east.
+ */
+
+#ifndef RECTANGLE_SURVEY_DEFAULT_SWEEP
+#define RECTANGLE_SURVEY_DEFAULT_SWEEP 25
+#endif
+
+#include "mcu_periph/uart.h"
+#include "messages.h"
+#include "subsystems/datalink/downlink.h"
+
+#if PERIODIC_TELEMETRY
+#include "subsystems/datalink/telemetry.h"
+#endif
+
+#include "firmwares/rotorcraft/navigation.h"
+
+#include "modules/nav/nav_survey_rectangle_rotorcraft.h"
+#include "state.h"
+
+float sweep = RECTANGLE_SURVEY_DEFAULT_SWEEP;
+
+static bool_t nav_survey_retange_active = FALSE;
```

Comment:
On 3 May 2015 at 19:53, Eduardo Reginato lavratti <notifications@github.com>
wrote:

> In sw/airborne/modules/nav/nav_survey_rectangle_rotorcraft.c
> <https://github.com/paparazzi/paparazzi/pull/1168#discussion_r29558855>:
>
> > +#include "mcu_periph/uart.h"
> > +#include "messages.h"
> > +#include "subsystems/datalink/downlink.h"
> > +
> > +#if PERIODIC_TELEMETRY
> > +#include "subsystems/datalink/telemetry.h"
> > +#endif
> > +
> > +#include "firmwares/rotorcraft/navigation.h"
> > +
> > +#include "modules/nav/nav_survey_rectangle_rotorcraft.h"
> > +#include "state.h"
> > +
> > +float sweep = RECTANGLE_SURVEY_DEFAULT_SWEEP;
> > +
> > +static bool_t nav_survey_retange_active = FALSE;
>
> Sorry Gautier ... what mean typo ?
>

He means retange should be rectangle.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/paparazzi/paparazzi/pull/1168/files#r29558855>.
>